cdap-api/src/main/java/co/cask/cdap/api/metadata/MetadataContext.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package co.cask.cdap.api.metadata;
+
+/**
+ * This interface provides methods that to create {@link MetadataEntity} for various resources present in CDAP.
+ * Additionally, this interface also includes methods to add metadata to various {@link MetadataEntity}.
+ */
+public interface MetadataContext {
+
+  /**
+   * Returns a metadata entity for the given dataset name present in the current namespace.
+   *
+   * @param datasetName the name of the dataset
+   * @return {@link MetadataEntity} for the given dataset in the current namespace
+   */
+  MetadataEntity getDatasetEntity(String datasetName);
```

Comment:
What's the use case of all these get method? Support this would not be straightforward since application runtime won't be able to talk to master in future.

Reply:
Or are these method for creating the object?? Shouldn't it call create?

Reply:
I am not a fan of having all these methods. When we generalize this in the near future, these will immediately become deprecated. Can't we have a method that creates a Dataset entity, and allow forming sub-entities from that? Like:

```
context.createEntity(EntityType.DATASET, "myDs").append("field", "salary");
```

and for now, we only support EntityType.DATASET, whereas in the future we can add more. We also have predefined qualifiers, such as:
```
context.createEntity(Qualifier.DATASET, "myDs").append(Qualifier.FIELD, "salary");
```
which would be defined for known things like schema fields, partitions, or files, but also allow passing in a string instead of Qualifier.CONSTANT.

Reply:
Yes they are just to create objects. I will rename them with create.